src/main/java/com/bina/varsim/tools/evaluation/VCFcompare.java

```diff
@@ -278,168 +290,224 @@ private void add_indels(List<Variant> var_list, int[] diff, byte[] ref, byte[][]
                     for (int j = 0; j < 2; j++) {
                       //recall: diff[j] = alt[j].length - ref.length
-                        if (i < ref.length + diff[j]) {
-                            idx[j] = i;
+                        if (relativePosition < referenceAlleleSequence.length + alleleLengthDifference[j]) {
+                            alleleSpecificRelativePosition[j] = relativePosition;
                         } else {
-                            idx[j] = -1; // we are into deleted bases
+                            alleleSpecificRelativePosition[j] = -1; // we are into deleted bases
                         }
                     }
                 } else {
                     for (int j = 0; j < 2; j++) {
-                        idx[j] = i + diff[j];
+                        //this one dictates that canonicalized variants will be located on the right end of reference allele
+                        alleleSpecificRelativePosition[j] = relativePosition + alleleLengthDifference[j];
                     }
                 }
 
-                if (idx[0] < 0 && idx[1] < 0) {
+                if (alleleSpecificRelativePosition[0] < 0 && alleleSpecificRelativePosition[1] < 0) {
                     // both deleted
-                } else if (idx[0] >= 0 && idx[1] < 0 && alt[0][idx[0]] != ref[i]) {
-                    // one deleted, hence the other is homozygous
+                    /*
+                    AGG
+                    --G
+                    --G
+                    ^    the scenario this if statement deals with
+                     */
+                } else if (alleleSpecificRelativePosition[0] >= 0 && alleleSpecificRelativePosition[1] < 0 &&
+                           alternativeAlleleSequence[0][alleleSpecificRelativePosition[0]] != referenceAlleleSequence[relativePosition]) {
+                    /* one deleted, hence the other is homozygous
+                    e.g. ref: AGG, alts: C,CT
+                    if we visualize the alignment, it looks like
+                    AGG
+                    -CT
+                    --C
+                     ^  the scenario this if statement deals with
+                    */
                     byte[] phase = {1, 1};
-                    var_list.add(new Variant(var.getChr(), curr_pos, 1, new byte[]{ref[i]},
-                            new FlexSeq[]{new FlexSeq(alt[0][idx[0]])}, phase, true, var.getVar_id(), ".", ""));
-                } else if (idx[0] < 0 && idx[1] >= 0 && alt[1][idx[1]] != ref[i]) {
+                    variantList.add(new Variant(variant.getChr(), currentPosition, 1, new byte[]{referenceAlleleSequence[relativePosition]},
+                            new FlexSeq[]{new FlexSeq(alternativeAlleleSequence[0][alleleSpecificRelativePosition[0]])}, phase, true, variant.getVariantId(), ".", ""));
```

Comment:
It would be good to use constants instead of things like "." to make it more understandable.